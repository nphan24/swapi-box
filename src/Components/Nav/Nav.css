.nav-link {
  margin: 20px;
}

.nav-div {
  margin-bottom: 20px;
}

.nav-link active {
  background-color: yellow;
}