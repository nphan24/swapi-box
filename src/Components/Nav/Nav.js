import React from 'react';
import { NavLink } from 'react-router-dom';
import './Nav.css';

const Nav = () => {
  return (
    <div>
      <button><NavLink to='./People'>People</NavLink></button>
      <button><NavLink to='./Planets'>Planets</NavLink></button>
      <button><NavLink to='./Vehicles'>Vehicles</NavLink></button>
    </div>
  );
};

export default Nav;