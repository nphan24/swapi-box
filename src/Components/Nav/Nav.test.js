import React from 'react';
import Nav from './Nav';
import { shallow } from 'enzyme';

describe('Nav', () => {
  let wrapper;
  let mockFunction;

  beforeEach(() => {
    mockFunction= jest.fn();

    wrapper = shallow(<Nav setInfo={mockFunction} />);
  });

  it('should exist', () => {
    expect(wrapper).toBeDefined();
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});