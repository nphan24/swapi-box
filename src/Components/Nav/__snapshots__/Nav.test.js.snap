// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Nav
    setInfo={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          className="nav-link"
          to="/"
>
          Home
</NavLink>,
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          className="nav-link"
          onClick={[Function]}
          to="/people"
>
          People
</NavLink>,
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          className="nav-link"
          onClick={[Function]}
          to="/planets"
>
          Planets
</NavLink>,
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          className="nav-link"
          onClick={[Function]}
          to="/vehicles"
>
          Vehicles
</NavLink>,
      ],
      "className": "nav-div",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": "Home",
          "className": "nav-link",
          "to": "/",
        },
        "ref": null,
        "rendered": "Home",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": "People",
          "className": "nav-link",
          "onClick": [Function],
          "to": "/people",
        },
        "ref": null,
        "rendered": "People",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": "Planets",
          "className": "nav-link",
          "onClick": [Function],
          "to": "/planets",
        },
        "ref": null,
        "rendered": "Planets",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": "Vehicles",
          "className": "nav-link",
          "onClick": [Function],
          "to": "/vehicles",
        },
        "ref": null,
        "rendered": "Vehicles",
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="nav-link"
            to="/"
>
            Home
</NavLink>,
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="nav-link"
            onClick={[Function]}
            to="/people"
>
            People
</NavLink>,
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="nav-link"
            onClick={[Function]}
            to="/planets"
>
            Planets
</NavLink>,
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="nav-link"
            onClick={[Function]}
            to="/vehicles"
>
            Vehicles
</NavLink>,
        ],
        "className": "nav-div",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": "Home",
            "className": "nav-link",
            "to": "/",
          },
          "ref": null,
          "rendered": "Home",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": "People",
            "className": "nav-link",
            "onClick": [Function],
            "to": "/people",
          },
          "ref": null,
          "rendered": "People",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": "Planets",
            "className": "nav-link",
            "onClick": [Function],
            "to": "/planets",
          },
          "ref": null,
          "rendered": "Planets",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": "Vehicles",
            "className": "nav-link",
            "onClick": [Function],
            "to": "/vehicles",
          },
          "ref": null,
          "rendered": "Vehicles",
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
